app/src/androidTest/java/com/example/android/architecture/blueprints/todoapp/test/chapter1/datainteraction/DataInteractionsTest.java

```diff
@@ -54,4 +59,57 @@ public void dataInteractionSample() {
                 .onChildView(withId(android.R.id.summary))
                 .check(matches(withText("sample@ema.il")));
     }
+
+    @Test
+    public void dataInteractionExercise() {
+        // Navigate to Settings
+        openDrawer();
+        onView(allOf(withId(R.id.design_menu_item_text),
+                withText(R.string.settings_title))).perform(click());
+
+        // Enter Notifications
+        onData(anything())
+                .inAdapterView(withId(android.R.id.list))
+                .atPosition(1)
+                .onChildView(withId(android.R.id.title))
+                .check(matches(withText("Notifications")))
+                .perform(click());
+
+        // Enable notifications
+        onData(withKey("notifications_new_message"))
+                .inAdapterView(allOf(withId(android.R.id.list), withParent(withId(android.R.id.list_container))))
+                .onChildView(allOf(withId(android.R.id.title), withText("Enable notifications")))
+                .perform(click());
+
+        if (isNotificationSwitchEnabled()) {
```

Comment:
In general, try to avoid conditional paths in tests. It would be way better to implement once scenario that test a flow where notification switches is disabled, and second one which test a flow where it is enabled (with additional checks)

Reply:
Yes I know that is why I have mentioned that in my PR description , normally notification switch by default is OFF so every time test in launched We know that switch is OFF . Probably because of my orchestrator which is removed (IDE +3.6 bug case) every time I launch single test then application starts with latest setup which means once enabled switch will be enabled for next runs so I need to find out his current state . I agree that there should be one test which test path when switch is enabled , provided changes.

Reply:
I see. But - this setting is actually a shared preference. How about writing a jUnit rule that will set this setting on/off before the test is started? We should always make sure that tests are independent, even if things like orchestrator doesn't work ;)

Reply:
sure , let me figure out how to deal with that then

Reply:
Done